common-content/en/module/tools/using-python-dependencies/index.md

```diff
@@ -0,0 +1,84 @@
++++
+title = "Using Python dependencies"
+headless = true
+time = 30
+facilitation = false
+emoji= "📖"
+objectives=[
+    "Write and run a Python program which uses third-party dependencies.",
+]
++++
+
+Let's create a small program which uses a dependency.
+
+We're going to use the `cowsay` library to make a program which outputs a picture of a cow saying something.
+
+First let's create a Python file which tries to use `cowsay`:
+
+{{<note>}}
+It's important that you don't name your Python file the same as the name of a library you're trying to import.
+
+We can't call our file `cowsay.py` because we're going to try to import `cowsay`.
+
+We can call it `main.py` or `hello.py` or `cow.py`. Just not `cowsay.py`.
+
+In this example, we'll call it `cow.py`.
+{{</note>}}
+
+```python
+import cowsay
+```
+
+If we try to run `python3 cow.py`, we'll get an error:
+
+```
+ModuleNotFoundError: No module named 'cowsay'
+```
+
+This is because we haven't installed cowsay yet.
+
+### Installing our dependency
+
+We will create a virtual environment, activate it, and install cowsay to it:
+
+```console
+% python3 -m venv .venv
+% . .venv/bin/activate
+(.venv) % echo cowsay > requirements.txt
+(.venv) % pip install -r requirements.txt
+```
+
+When we activate a virtual environment, its name gets shown before our terminal prompt. This is a useful reminder that we're in a virtual environment!
+
+### Running our program
+
+Now if we run `python3 cow.py` we don't get an error - we installed `cowsay` into our active virtual environment.
+
+If we open a new terminal and run `python3 cow.py` we'll get an error again. Because we haven't activated a virtual environment.
```

Comment:
```suggestion
Open a new terminal and run `python3 cow.py`. You will get an error again! This is because we haven't activated a virtual environment.
```